### PR TITLE
Restructure help dialog

### LINKED ISSUE #384: Restructure editor help dialog
**Which part of the project should be enhanced?**

The help dialog in the editor.

**Is your enhancement request related to a problem? Please describe.**

Not directly.

**Describe the solution you'd like**

The help dialog should include the keybindings of the various functions, have the correct links for source code etc. (remember the frontend resides in another repo than the backend), and maybe the cheatsheet could be more structured or better styled.

**Describe alternatives you've considered**

Add tabs into the help dialog for keybindings, cheatsheet and further links.

**Additional context**

#100

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
help dialog

### Description
This PR improves the help dialog.

### Steps

- [x] added implementation
- [ ] added / updated tests
- [ ] added / updated documentation
- [x] extended changelog

### Related Issue(s)
fixes #384

### CODE REVIEW

File: src/components/editor/app-bar/help-button/help-button.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        return (<Cheatsheet/>)
+      case HelpTabStatus.Shortcuts:
+        return (<Shortcut/>)
+      case HelpTabStatus.Links:
+        return (<Links/>)
+    }
+  }
+
+  const handleShow = () => setShow(true)
+  const handleClose = () => setShow(false)
```
Comment (ErikMichelson) [inline src/components/editor/app-bar/help-button/help-button.tsx:32]: Both functions are called just one time. Maybe consider inlining them?

File: src/components/editor/app-bar/help-button/links.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                className='text-primary'
+              />
+            </li>
+            <li>
+              <TranslatedExternalLink
+                i18nKey='editor.help.documents.yamlMetadata'
+                href='/n/yaml-data'
+                icon='dot-circle-o'
+                className='text-primary'
+              />
```
Comment (ErikMichelson) [inline src/components/editor/app-bar/help-button/links.tsx:68]: These are internal links and not external links.

**Review by ErikMichelson — CHANGES_REQUESTED** [review]
Some code reformatting would be nice. There are some lines with a single closing JSX-bracket that could be placed in the line above etc.

~~And more important question: Where do all these removals in the yarn.lock originate from?~~

Please rebase from latest main.

**Comment by mrdrogdrog** [comment]
Please rebase

File: src/components/editor/app-bar/help-button/cheatsheet.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  return (
+    <Table className="table-condensed table-cheatsheet">
+      <thead>
+        <tr>
+          <th><Trans i18nKey='editor.help.cheatsheet.example'/></th>
+          <th><Trans i18nKey='editor.help.cheatsheet.syntax'/></th>
+        </tr>
+      </thead>
+      <tbody className="markdown-body"
+        style={{ fontFamily: 'inherit', fontSize: '14px', padding: 0, maxWidth: 'inherit' }}>
```
Comment (mrdrogdrog) [inline src/components/editor/app-bar/help-button/cheatsheet.tsx:39]: - this tbody doesn't need the markdown-body.
- static style

File: src/components/editor/app-bar/help-button/cheatsheet.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        {codes.map((code, i) => {
```
Comment (mrdrogdrog) [inline src/components/editor/app-bar/help-button/cheatsheet.tsx:40]: `i`

**Review by mrdrogdrog — CHANGES_REQUESTED** [review]

File: src/components/editor/app-bar/help-button/cheatsheet.tsx (hunk 3 above)
Comment (DerMolly) [reply src/components/editor/app-bar/help-button/cheatsheet.tsx:39]: done

File: src/components/editor/app-bar/help-button/cheatsheet.tsx (hunk 4 above)
Comment (DerMolly) [reply src/components/editor/app-bar/help-button/cheatsheet.tsx:40]: done

File: src/components/editor/app-bar/help-button/links.tsx (hunk 2 above)
Comment (DerMolly) [reply src/components/editor/app-bar/help-button/links.tsx:68]: done

File: src/components/editor/app-bar/help-button/help-button.tsx (hunk 1 above)
Comment (DerMolly) [reply src/components/editor/app-bar/help-button/help-button.tsx:32]: done

**Review by mrdrogdrog — APPROVED** [review]

**Review by mrdrogdrog — APPROVED** [review]

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**